prover/src/matrix.rs

```diff
@@ -22,6 +22,7 @@ use utils::iterators::*;
 /// - A matrix must consist of at least 1 column and at least 2 rows.
 /// - All columns must be of the same length.
 /// - Number of rows must be a power of two.
+#[derive(Debug, Clone)]
 pub struct Matrix<E: FieldElement> {
```

Comment:
Outside of this PR scope, but have you considered using a single vector for efficiency? If it can bring some speedups, then this structure could also be used at some other places in the library where number of columns/rows are known from the start.

Reply:
I'm hoping that in the future we can make `Matrix` into a trait. This way, users would be able to define the most suitable underlying structure which could be something as complex as memory-mapped objects (e.g. https://github.com/novifinancial/rust-plasma/tree/main/plasma-store).

Reply:
That would be great indeed! Looking forward to it!